code/model/DynamoDbSession.php

```diff
@@ -40,14 +42,20 @@ public static function get()
         if (defined('AWS_DYNAMODB_SESSION_TABLE') && AWS_DYNAMODB_SESSION_TABLE) {
             $dynamoOptions = array('region' => AWS_REGION_NAME);
 
+           // This endpoint can be set for locally testing DynamoDB.
+           // see http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.html
+           if (defined('AWS_DYNAMODB_ENDPOINT')) {
+                $dynamoOptions['endpoint'] = AWS_DYNAMODB_ENDPOINT;
+           }
+
             if (defined('AWS_ACCESS_KEY') && defined('AWS_SECRET_KEY')) {
-                $dynamoOptions['key'] = AWS_ACCESS_KEY;
-                $dynamoOptions['secret'] = AWS_SECRET_KEY;
+                $dynamoOptions['credentials']['key'] = AWS_ACCESS_KEY;
+                $dynamoOptions['credentials']['secret'] = AWS_SECRET_KEY;
             } else {
                 // cache credentials when IAM fetches the credentials from EC2 metadata service
-                // this will use doctrine/cache (included via composer) to do the actual caching into the filesystem
+                // this will use doctrine/cache (included via composer) to do the actual caching into APCu
                 // http://docs.aws.amazon.com/aws-sdk-php/guide/latest/performance.html#cache-instance-profile-credentials
-                $dynamoOptions['credentials.cache'] = true;
+                $dynamoOptions['credentials'] = new DoctrineCacheAdapter(new ApcuCache());
```

Comment:
Is it possible to fallback to PhpFileCache or ArrayCache cache instead of require the module?